src/transformers/models/auto/auto_factory.py

```diff
@@ -431,12 +432,17 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
         ]
         hub_kwargs = {name: kwargs.pop(name) for name in hub_kwargs_names if name in kwargs}
         if not isinstance(config, PretrainedConfig):
+            kwargs_copy = copy.deepcopy(kwargs)
+            # ensure not to pollute the config object with torch_dtype="auto" - since it's meaningless in the context of the config object - torch.dtype values are acceptable
+            if kwargs_copy.get("torch_dtype", None) == "auto":
+                _ = kwargs_copy.pop("torch_dtype")
+
             config, kwargs = AutoConfig.from_pretrained(
                 pretrained_model_name_or_path,
                 return_unused_kwargs=True,
                 trust_remote_code=trust_remote_code,
                 **hub_kwargs,
-                **kwargs,
+                **kwargs_copy,
```

Comment:
While doing testing I found there was a bug here. `kwargs` of model's `from_pretrained` was overriding config object wrt to `torch_dtype="auto"` which is very wrong.

Except when passing an actual dtype like `torch_dtype=torch.float16` I think it should be reflected in the config object, correct?

this is how it's currently implemented and a test added that covers the original bug.